driver-core/src/test/unit/com/mongodb/client/model/search/SearchOperatorsTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.search;
+
+import org.bson.BsonBoolean;
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonString;
+import org.junit.jupiter.api.Test;
+
+import static com.mongodb.client.model.search.SearchPath.path;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SearchOperatorsTest {
+    @Test
+    public void testTextSearchOperator() {
```

Comment:
This gives a flavor of the unit tests we should have for all the options of all the operators.  The assertions are that each operator produce the correct BSON document.   I start with an assertion based on only the required values, then I configure all the options and assert again.  I think this should cover all the null checks required in `toBsonDocument`.